client/components/domains/register-domain-step/index.jsx

```diff
@@ -541,10 +539,20 @@ const RegisterDomainStep = React.createClass( {
 				severity = 'info';
 				break;
 			case 'not_registrable':
-				if ( domain.indexOf( '.' ) ) {
-					message = this.translate( 'Sorry but %(domain)s cannot be registered on WordPress.com.', {
-						args: { domain: domain }
-					} );
+				const tldIndex = domain.lastIndexOf( '.' );
```

Comment:
that's a good point since .co.uk results in 
<img width="790" alt="screen shot 2016-08-05 at 9 56 28 am" src="https://cloud.githubusercontent.com/assets/8921630/17444093/041c301a-5af3-11e6-9abe-b9b742196fdb.png">